wpimath/src/main/java/edu/wpi/first/wpilibj/waypoint/WaypointUtil.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj.waypoint;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.geometry.Translation2d;
+import edu.wpi.first.wpilibj.spline.Spline.ControlVector;
+import edu.wpi.first.wpilibj.trajectory.Trajectory;
+import edu.wpi.first.wpilibj.trajectory.TrajectoryConfig;
+import edu.wpi.first.wpilibj.trajectory.TrajectoryGenerator;
+import edu.wpi.first.wpilibj.trajectory.TrajectoryGenerator.ControlVectorList;
+
+public final class WaypointUtil {
+  private static final Logger LOGGER = Logger.getLogger(WaypointUtil.class.getName());
+
+  private WaypointUtil() {
+    throw new UnsupportedOperationException("This is a utility class!");
+  }
+
+  /**
+   * Imports Path object and creates a trajectory with that info.
+   *
+   * @param filepath Full path to the file
+   * @param config TrajectoryConfig that sets the contraints for the Trajectory to be generated
+   *
+   * @return Trajectory created from the waypoints in the file.
+   */
+  public static Trajectory importPathToCubicTrajectory(String filepath, TrajectoryConfig config) {
+    try (FileReader fr = new FileReader(new File(filepath));
+      BufferedReader reader = new BufferedReader(fr)) {
+      
+      var interiorWaypoints = new ArrayList<Translation2d>();
+      Pose2d start = new Pose2d();
+      Pose2d end = new Pose2d();
+      int loop = 0;
+      String line;
+      String lastline = "";
+
+      while ((line = reader.readLine()) != null) {
+        if (loop == 0) {
+          if (line != "X,Y,Tangent X,Tangent Y,Fixed Theta,Reversed,Name") {
+            throw new RuntimeException("this isn’t a PathWeaver csv file");
+          }  
+        }
+        else if (loop == 1) {
+          start = createPoseWaypoint(line);
+        }
+        else if (loop == 2) {
+          ; // skip the second line because we are logging last
+        }
```

Comment:
why are we skipping the second waypoint?

Reply:
we are logging one behind because we don't know where the end of the file is.  If the row is the last row then we log it as the end Pose.  At 2 we don't want to relog the start pose.